tools/integration-tests/tester/tests/testutil.go

```diff
@@ -619,3 +619,52 @@ func txMetadataStateEqual(t *testing.T, node *framework.Node, txID string, expIn
 	}
 	return true, metadata.GradeOfFinality
 }
+
+// TryConfirmMessage tries to confirm the message on all the peers provided within the time limit provided.
+func TryConfirmMessage(t *testing.T, n *framework.Network, requiredPeers []*framework.Node, msgID string, waitFor time.Duration, tick time.Duration) {
+	var peers []*framework.Node
+	for _, peer := range n.Peers() {
```

Comment:
Is there a use-case to collect the peers on which the message is available separately if we want the requiredPeers to have the message confirmed?

Reply:
yeah. It's a short circuit check to make sure that the message exists in the network in the first place.
Else, you'll wait for the whole timeout for nothing.